spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/async/TraceDelegate.java

```diff
@@ -24,7 +24,7 @@
  */
 public abstract class TraceDelegate<T> {
```

Comment:
is this type used anywhere except from a runnable or a callable? If so, I'd probably flatten the type hierarchy to just runnable, callable as the types aren't interchangeable anyway, and the work here isn't high value.